src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -162,6 +160,14 @@ public OAuthHandler(HttpClient backchannel)
             var ticket = new AuthenticationTicket(new ClaimsPrincipal(identity), properties, Options.AuthenticationScheme);
             var context = new OAuthCreatingTicketContext(ticket, Context, Options, Backchannel, tokens);
             await Options.Events.CreatingTicket(context);
+
+            if (ticket.Principal.FindFirst(ClaimTypes.NameIdentifier) == null)
+            {
+                // If there aren't any claims at all, at least add a unique id for this login session. This will help associate the
+                // tokens from the token store.
+                identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, Guid.NewGuid().ToString(), ClaimValueTypes.String, Options.ClaimsIssuer));
```

Comment:
Yeah, this scenario is pretty strange. Apps using the middleware without at least getting a user id seems to defeat the "Authentication" purpose of the middleware. "This is some google user, though we couldn't tell you which one...".

Other suggestions for associating this psudo-anonymous user and their tokens?